internal/server/handlers.go

```diff
@@ -22,13 +24,62 @@ import (
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/internal/server/providers"
 	"github.com/infrahq/infra/internal/server/redis"
+	"github.com/infrahq/infra/uid"
 )
 
 type API struct {
 	t          *Telemetry
 	server     *Server
-	migrations []apiMigration
 	openAPIDoc openapi3.T
+	versions   map[routeIdentifier][]routeVersion
+}
+
+type routeVersion struct {
+	version *semver.Version
+	handler func(c *gin.Context)
+}
+
+// nolint // will be used in the future
+func addVersionHandler[Req, Res any](a *API, method, path, version string, routeDef route[Req, Res]) {
```

Comment:
What is this going to be used for?

Reply:
Oops, this function is already used in this PR! I used it for the one api migration we have today (just below here). I forgot to remove this comment. I've done that now in the latest commit.